src/main/java/duck/Parser.java

```diff
@@ -0,0 +1,252 @@
+            String description = firstPart.length > 1 ? firstPart[1] : "";
+            String from = parts.length > 1 ? parts[1].trim() : "";
+            String to = parts.length > 2 ? parts[2].trim() : "";
+            if (description.isBlank() || from.isBlank() || to.isBlank()) {
+                throw new EmptyDetailsException("No description or 'from' or 'to' time provided");
+            }
+            Task task = new Event(false, description, from, to);
+            list.add(task);
+            return ui.addTaskMessage(task, list);
+
+        } catch (EmptyDetailsException e) {
+            return ui.showErrorMessage(e.getMessage());
+        } catch (Exception e) {
+            return ui.showErrorMessage("Invalid date format! Use yyyy-MM-dd HHmm (e.g., 2025-01-27 2359).");
+        }
+    }
+
+    /**
+     * Deletes a task from the list.
+     *
+     * @param userInput The user input containing task number.
+     * @param list The task list containing the task.
+     * @param ui The UI instance to display feedback.
+     */
+    public String delete(String userInput, TaskList list, Ui ui) {
+        try {
+            String[] input = userInput.split(" ");
+            int number = Integer.parseInt(input[1]);
+            if (number <= 0 || number > list.size()) {
+                throw new IndexOutOfBoundsException("Task number out of range.");
+            }
+            Task task = list.remove(number - 1);
+            return ui.removeTaskMessage(task, list);
+        } catch (IndexOutOfBoundsException e) {
+            return ui.showErrorMessage(e.getMessage());
+        } catch (Exception e) {
+            return ui.showErrorMessage("Invalid Input, Try again!");
+        }
+    }
+
+    /**
+     * Displays a matching list of tasks based on keyword
+     *
+     * @param userInput The user input containing keyword
+     * @param list The task list containing the task.
+     * @param ui The UI instance to display feedback.
+     */
+    public String find(String userInput, TaskList list, Ui ui) {
+        try {
+            String[] input = userInput.split(" ");
+            String keyword = input.length > 1 ? input[1] : "";
+            if (keyword.isBlank()) {
+                throw new EmptyDetailsException("No keyword provided");
+            }
+            TaskList match = list.find(keyword);
+            return ui.findMessage(match);
+        } catch (Exception e) {
+            return ui.showErrorMessage(e.getMessage());
+        }
+    }
+
+
+    /**
+     * Displays the goodbye message.
+     *
+     * @param ui The UI instance to display the message.
+     */
+    public String bye(Ui ui) {
+        return ui.showByeMessage();
+    }
+
+    /**
+     * Displays an error message for unknown commands.
+     *
+     * @param ui The UI instance to display the message.
+     */
+    public String unknown(Ui ui) {
+        return ui.showCommandErrorMessage();
+    }
```

Comment:
Should more descriptive method names be used? I would recommend `handleXXXCommand` 

```suggestion
    public String handleUnknownCommand(Ui ui) {
        return ui.showCommandErrorMessage();
    }
```

Reply:
Thanks for the suggestion 👍🏻